Core/Core/Configuration/Config/AgreementConfig.swift

```diff
@@ -10,17 +10,55 @@ import Foundation
 private enum AgreementKeys: String {
     case privacyPolicyURL = "PRIVACY_POLICY_URL"
     case tosURL = "TOS_URL"
+    case cookiePolicyURL = "COOKIE_POLICY_URL"
+    case dataSellContentURL = "DATA_SELL_CONSENT_URL"
+    case supportedLanguages = "SUPPORTED_LANGUAGES"
 }
 
 public class AgreementConfig: NSObject {
     public var privacyPolicyURL: URL?
     public var tosURL: URL?
-    
+    public var cookiePolicyURL: URL?
+    public var dataSellContentURL: URL?
+    public var supportedLanguages: [String]?
+
     init(dictionary: [String: AnyObject]) {
-        privacyPolicyURL = (dictionary[AgreementKeys.privacyPolicyURL.rawValue] as? String).flatMap(URL.init)
-        tosURL = (dictionary[AgreementKeys.tosURL.rawValue] as? String).flatMap(URL.init)
+        supportedLanguages = dictionary[AgreementKeys.supportedLanguages.rawValue] as? [String]
+        cookiePolicyURL = (dictionary[AgreementKeys.cookiePolicyURL.rawValue] as? String).flatMap(URL.init)
+        dataSellContentURL = (dictionary[AgreementKeys.dataSellContentURL.rawValue] as? String).flatMap(URL.init)
+
         super.init()
+
+        if let tosURL = dictionary[AgreementKeys.tosURL.rawValue] as? String {
+            self.tosURL = URL(string: completePath(url: tosURL))
+        }
+
+        if let privacyPolicyURL = dictionary[AgreementKeys.privacyPolicyURL.rawValue] as? String {
+            self.privacyPolicyURL = URL(string: completePath(url: privacyPolicyURL))
+        }
     }
+
+    private func completePath(url: String) -> String {
+           let langCode = String(Locale.preferredLanguages.first?.prefix(2) ?? "")
```

Comment:
Hardcoding the prefix isn't a good option. I guess you can use the following code to get the desired code
```
let langCode: String
        if #available(iOS 16, *) {
            langCode = Locale.current.language.languageCode?.identifier ?? ""
        } else {
            langCode = Locale.current.languageCode ?? ""
        }
```